backend/app/api/endpoints/keys.py

```
@@ -0,0 +1,31 @@
+async def update_keys(keys: APIKeyUpdate):
+    """Set API keys used by the application"""
+    try:
+        return api_key_service.set_keys(
+            coingecko_api_key=keys.coingecko_api_key,
+            binance_api_key=keys.binance_api_key,
+            binance_api_secret=keys.binance_api_secret,
+            news_api_key=keys.news_api_key,
+        )
+    except Exception as e:
+        logger.error(f"Error setting API keys: {e}")
+        raise HTTPException(status_code=500, detail=str(e))
```

Ruff (0.11.9)

Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)